### ISSUE TITLE
node_redis: WARNING: You passed "rediss" as protocol instead of the "redis" protocol!

### ISSUE THREAD

**Issue by siluri** [NodeRedis/node-redis#1331]
`node_redis: WARNING: You passed "rediss" as protocol instead of the "redis" protocol!`

this warning is displayed because `rediss:` is checked but `rediss` would be correct - [Line 35](https://github.com/NodeRedis/node_redis/blob/1a4c410b4179d68aee922b80aff583a336251163/lib/createClient.js)

**Comment by coderlongren** [comment]
rediss is just wrong way because the commond do

**Comment by BridgeAR** [comment]
The message should indeed be updated but it seems like you somehow passed through a wrong protocol.

Can you please show what you passed to `Redis.createClient()` as args (replace username / password / concrete urls) and keep everything exactly as it was passed to node_redis?

**Comment by istvanp** [comment]
I encountered the same issue. Then I realized the commit enabling `rediss` as a protocol is not released yet. The latest version, `2.8.0`, does not have this commit yet: https://github.com/NodeRedis/node_redis/commit/5d6e471d0c14df39974000427719c5d85fa4f21f

**Comment by ashishth09** [comment]
When is the new release? I need this feature.

**Comment by Teamop** [comment]
@BridgeAR @stockholmux @Salakar friendly ping. Maybe a new release is needed for this warning.

**Comment by Salakar** [comment]
@Teamop unfortunately I don't have access to publish anything to npm - only @BridgeAR  can.

**Comment by coguy450** [comment]
I ran into this issue. If look at the online repo, I see the statement block for rediss.
https://github.com/NodeRedis/node_redis/blob/23ef1e7afadd6022526860d22ac0a7f78495106d/lib/createClient.js#L34

However, when I examine my actual node_modules folder, the version says it's 2.8, however the createClient file does not have that rediss statement block. I fixed locally by pasting the rediss if statement into my local node modules and that worked. I'm not sure how the distributed version is different than what's shown online in the repo.
![image](https://user-images.githubusercontent.com/8247495/45722027-c66f2b00-bb67-11e8-93f6-89e5bf431c72.png)

**Comment by bluesxman** [comment]
@BridgeAR What we're pulling down for 2.8.0 doesn't actually support TLS (i.e. a url starting with "rediss:").  That's the real problem; the warning is just a symptom.

**Comment by roschler** [comment]
Is there any major value to using the TLS support?

**Comment by willks** [comment]
Bah! Project is dead? 1327 forks. No TLS for me - yet another fork. Ahh don't you just love NodeJs?

**Comment by jasonamyers** [comment]
I've love to have it today, it would let us talk to encrypted auth AWS ElasticCache

**Comment by willks** [comment]
@jasonamyers I had more success with https://github.com/luin/ioredis on this issue - but you need a cachecluster setup, with at rest encryption, transport encryption and a password. API's are similar.

**Comment by scriptonian** [comment]
Hello can someone please advice when this is going into master and getting released? Having this issue and had to manually overwrite createClient.js in node_module. This issue has been opened for over a year. Can we push this out and allow TLS ?

**Comment by venkatesh-nadar** [comment]
Currently, this section

```js
if (parsed.protocol && parsed.protocol !== 'redis:') {
    console.warn('node_redis: WARNING: You passed "' + parsed.protocol.substring(0, parsed.protocol.length - 1) + '" as protocol instead of the "redis" protocol!');
}
```
need to be replaced by 

```js
if (parsed.protocol === 'rediss:') {
    options.tls = options.tls || {};
} else if (parsed.protocol !== 'redis:') {
    console.warn('node_redis: WARNING: You passed "' + parsed.protocol.substring(0, parsed.protocol.length - 1) + '" as protocol instead of the "redis" protocol!');
}
```
in createClient.js inorder to get **rediss://** support or you could try another way like shown below.

```js
var redis = require("redis");
var client = redis.createClient("redis://:password@host:port", {tls:{}});
```

By passing tls as empty object got my redis client got connected without any issue.

@scriptonian this should also solve your issue by avoiding manual overwrite in createClient.js

**Please Note :** 
Bare me if my commenting style is odd. I am new in github. :)

**Comment by scriptonian** [comment]
thank you venkatesh, yes i had already made this change as coguy450 had suggested it in the comments. but i am surprised the team hasnt updated this. Its been over a year !

**Comment by Salakar** [comment]
> but i am surprised the team hasnt updated this. Its been over a year !

I have reached out to the current owner of the npm package multiple times over the past year re ownership - as our company (@invertase) is willing to sponsor the resources (contributing and project management) to keep this project active - as we rely on it for some legacy projects; but I've never had any response on the subject, unfortunately. 

I would like to begin regular release cycles again for the library and I have contributed to it in the past several times including to core functionality (such as the `redis-parser`) - so I think I have a fairly good understanding of the code base. 

I currently only have partial access to the repo/org so can merge and add code in some repos; but unfortunately cannot publish anything to npm anyway.

Maybe `ioredis` would be a better one to start switching to / focus on (we use it for all new projects) given that this one is practically abandoned? @luin does a really great job keeping it actively maintained 🤗

**Comment by bmudda** [comment]
+1

**Comment by nfedyk** [comment]
is there a new version coming with this fix?

**Comment by peterpoliwoda** [comment]
Somebody wants to make a pull request to support the rediss:// url?

**Comment by venkatesh-nadar** [comment]
@peterpoliwoda it seems to be already added in the master branch... 👍

**Comment by peterpoliwoda** [comment]
Hm... yeah I see it’s in the repo but npm doesn’t install with the fix. The version number hasn’t changed so ‘npm install’ keeps the old build, like it had an old tarball in the archive. 
I got it working by installing directly from Github with
‘’’
npm install https://github.com/NodeRedis/node_redis
‘’’

Thanks a ton @venkatesh-nadar!

**Comment by IdanAdar** [comment]
@BridgeAR Can we get a new release in npm please?

**Comment by Salakar** [comment]
Have reached out a few times to @BridgeAR requesting npm publish access over the past year and nothing has happened out of it, I've given up trying to help maintain this repo for the time being. Your best bet is switching to [ioredis](https://github.com/luin/ioredis), @luin is doing a great job keeping it maintained.

**Comment by Salakar** [comment]
Hey everyone, I've taken over as lead maintainer and have all the required access now 🎉

I've spent the last few days getting master ready for a release, and as of a few minutes ago I've just published v3.0.0 to NPM; https://github.com/NodeRedis/node-redis/releases/tag/v3.0.0 - which includes this change.

Expect regular releases - my first priority at the moment is making this project more contributor friendly to ensure that the project lives and continues to grow and is not blocked by any singular person's time. To do this I'd like to grow a larger set of shallow contributors. With this I hope to mitigate the previous problem of the project needing updates but there being no-one who has the power to do so. I'm working through the following on this;

 - [x] Contributing docs & Code of Conduct
 - [x] Setup an [Open Collective](https://opencollective.com/node-redis) and a contributors [expenses policy](https://opencollective.com/node-redis/expenses/new)
   - You'll notice the shiny new `Sponsor` button at the top of GitHub, I've also gone ahead and sponsored it myself and through my company to help kick start it for any future contributors
 - **WIP:** Release automation & semantic versioning (publishing to NPM, generating changelogs, etc)
 - **WIP:** Improve CI, e.g. Windows CI is super slow & flakey right now

After that I'll be switching my attention to modernising (e.g. promises) & clearing technical debt in the Node Redis codebase. @BridgeAR has already done a boat load of stuff for this, if you're curious check out the WIP [v4 branch](https://github.com/NodeRedis/node-redis/tree/v4) and its changelog.
